pkg/library/libraryinputresources/resource.go

```diff
@@ -85,9 +85,21 @@ func ResourceFromFile(location, fileTrimPrefix string) (*Resource, error) {
 	retFilename := strings.TrimPrefix(location, fileTrimPrefix)
 	retFilename = strings.TrimPrefix(retFilename, "/")
 
+	// If the file represents a list, derive the resource name from the file name.
+	// Otherwise, if it represents a single resource, use the parent directory name instead.
+	resourceContent := ret.(*unstructured.Unstructured)
+	resourceName := filepath.Base(filepath.Dir(location))
+	if resourceContent.IsList() {
+		resourceName = strings.TrimSuffix(filepath.Base(location), ".yaml")
+	}
 	return &Resource{
+		ResourceType: schema.GroupVersionResource{
+			Group:    ret.GetObjectKind().GroupVersionKind().Group,
+			Version:  ret.GetObjectKind().GroupVersionKind().Version,
+			Resource: resourceName,
```

Comment:
If a resource is individual (reasonable), then this method should return a slice of resources with proper names rather than a list with an incorrect name